tools/plugins/com.liferay.ide.project.ui/src/com/liferay/ide/project/ui/dialog/LiferayServiceProjectSelectionDialog.java

```diff
@@ -54,8 +53,22 @@ protected boolean checkProject( IJavaProject javaProject )
         }
     }
 
-    public void setProjects( List<IProject> projects )
+    private List<IProject> getServiceBuilderProjects()
     {
-        this.projects = projects;
+        List<IProject> results = new ArrayList<IProject>();
+
+        IProject[] projects = CoreUtil.getAllProjects();
+
+        for( IProject project : projects )
+        {
+            IFile serviceFile = project.getFile( "/docroot/WEB-INF/service.xml" );
```

Comment:
hey Greg , now all of the logic in code upgrade tool is just for 62-sdk type projects , such as build service page and compile page , we just directly call related command in ant command. So if you will I can change the logic to adapt to sdk and maven project.(via IProjectBuilder), but what about complile page ? just compile sdk project or maven also ?